app/controllers/challenges_controller.rb

```diff
@@ -42,6 +43,27 @@ def submit_for_approval
     end
   end
 
+  # rubocop:disable Metrics/MethodLength
+  def approve
+    challenge = Challenge.find(params[:id])
+
+    if challenge.update(stage: :done)
+      if challenge.performance.challenges.all?(&:done_stage?)
+        switcher = Challenge::SpotSwitcher.new(challenge.performance)
+        begin
+          switcher.run!
```

Comment:
This looks (from a brief glance) like it might slow down the request a lot w/ all of those database hits. Maybe consider looking into something like [sidekiq](https://github.com/mperham/sidekiq) to handle this in the background. They have some [docs](https://github.com/mperham/sidekiq/wiki/Deployment#heroku) on getting things set up with Heroku.

Reply:
That's the plan. I'm just gonna do this at a later time